File: arkts_static_test/03_Types/ets_cases/3.13_Type_string/cross_lang_verify/StringConcatTest.swift
/*
* Copyright (c) 2021-2026 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/
/**
 * Swift 版本 - 字符串连接与插值测试
 * 验证 string 与其他类型的连接
 */

import Foundation

// 1. string + int (使用字符串插值)
let i: Int = 42
let si = "i=\(i)"
assert(si == "i=42", "int concat failed: \(si)")

// 2. string + Int64 (类似 long)
let l: Int64 = 1000000000
let sl = "l=\(l)"
assert(sl == "l=1000000000", "long concat failed: \(sl)")

// 3. string + Float
let f: Float = 3.14
let sf = "f=\(f)"
assert(sf.hasPrefix("f=3.14"), "float concat failed: \(sf)")

// 4. string + Double
let d: Double = 2.718
let sd = "d=\(d)"
assert(sd.hasPrefix("d=2.718"), "double concat failed: \(sd)")

// 5. string + Bool
let b1: Bool = true
let b2: Bool = false
let sb1 = "b1=\(b1)"
let sb2 = "b2=\(b2)"
assert(sb1 == "b1=true", "boolean true concat failed")
assert(sb2 == "b2=false", "boolean false concat failed")

// 6. string + nil (Optional)
let nullable: String? = nil
let sn = "n=\(nullable ?? "nil")"
assert(sn == "n=nil", "nil concat failed: \(sn)")

// 7. number + string
let ns = "\(42) is the answer"
assert(ns == "42 is the answer", "number + string failed")

// 8. 字符串插值表达式
let expr = "2 + 3 = \(2 + 3)"
assert(expr == "2 + 3 = 5", "expression interpolation failed")

print("All Swift concat tests passed!")

File: arkts_static_test/03_Types/ets_cases/3.13_Type_string/cross_lang_verify/StringConcatTest.java
/**
 * Java 版本 - 字符串连接与隐式转换测试
 * 验证 string + number, string + boolean, string + null 等
 */
public class StringConcatTest {
    public static void main(String[] args) {
        // 1. string + int
        int i = 42;
        String si = "i=" + i;
        assert si.equals("i=42") : "int concat failed: " + si;
        
        // 2. string + long
        long l = 1000000000L;
        String sl = "l=" + l;
        assert sl.equals("l=1000000000") : "long concat failed: " + sl;
        
        // 3. string + float
        float f = 3.14f;
        String sf = "f=" + f;
        // Java 浮点数转换可能有精度问题
        assert sf.startsWith("f=3.14") : "float concat failed: " + sf;
        
        // 4. string + double
        double d = 2.718;
        String sd = "d=" + d;
        assert sd.startsWith("d=2.718") : "double concat failed: " + sd;
        
        // 5. string + boolean
        boolean b1 = true;
        boolean b2 = false;
        String sb1 = "b1=" + b1;
        String sb2 = "b2=" + b2;
        assert sb1.equals("b1=true") : "boolean true concat failed";
        assert sb2.equals("b2=false") : "boolean false concat failed";
        
        // 6. string + null
        String nullable = null;
        String sn = "n=" + nullable;
        assert sn.equals("n=null") : "null concat failed: " + sn;
        
        // 7. number + string (左侧是数值)
        String ns = 42 + " is the answer";
        assert ns.equals("42 is the answer") : "number + string failed";
        
        System.out.println("All Java concat tests passed!");
    }
}
